cinema/views.py

```diff
@@ -1 +1,76 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import (
+    Order,
+    Ticket,
+    CinemaHall,
+    Actor,
+    Genre,
+    Movie,
+    MovieSession
+)
+from cinema.serializers import (
+    CinemaHallSerializer,
+    ActorSerializer,
+    GenreSerializer,
+    MovieSerializer,
+    MovieSessionSerializer,
+    TicketSerializer,
+    OrderSerializer,
+    MovieListSerializer,
+    MovieDetailSerializer,
+    MovieSessionListSerializer,
+    MovieSessionDetailSerializer,
+    CinemaHallDetailSerializer,
+)
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+
+    def get_serializer_class(self):
+        if self.action == "retrieve":
+            return CinemaHallDetailSerializer
+        return CinemaHallSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieDetailSerializer
+        return MovieSerializer
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all()
```

Comment:
You don't need to use `.all()` before `.prefetch_related()` or `.select_related()`. Consider removing `.all()` if you plan to use these methods.